README.md

```diff
@@ -66,3 +66,77 @@ To support browsers that don't have built-in support for the Web Components API,
 import "@webcomponents/custom-elements";
 import "@mercurymedia/elm-ag-grid";
 ```
+
+## Custom views for cells
+
+Custom cell content (such as a button or link) can be created and used as a separate Elm application. This application must be defined as `Browser.element` in load it into the table cell.
+
+FYI, there are two components in the [example application](https://github.com/mercurymedia/elm-ag-grid/tree/main/examples) used that you might find very useful.
+
+### Register Component
+
+To reference the component application and use it for the table, you must register the component in the `window.ElmAgGridComponentRegistry` object. The key is the name used in your main application to refer to the component and the value is the object containing the `init` function to initialize the Elm application.
+
+You can either just use the usual Elm object for this:
+
+```javascript
+import { Elm } from "./src/Components/Link.elm";
+
+window.ElmAgGridComponentRegistry = {
+  linkRenderer: Elm.Components.Link,
+};
+```
+
+Or define your own object with an `init` function if you want to use ports for communication between the applications.
+
+**Note:** In this case, make sure to return the initialized application to be able to access the ports from the cellRenderer.
+
+```javascript
+import { Elm } from "./src/Components/Button.elm";
+
+window.ElmAgGridComponentRegistry = {
+  buttonRenderer: {
+    init: function ({ node, flags }) {
+      let component = Elm.Components.Button.init({
+        node: node,
+        flags: flags,
+      });
+
+      component.ports.onButtonClick.subscribe(function (id) {
+        // This `app` is the main application
+        if (app) app.ports.buttonClicked.send(id);
+      });
+
+      return component;
+    },
+  },
+};
+```
+
+**The function is called with an object containing the keys `node` and `flags`. The flags are containing the usual [AgGrid cellRendererParams](https://www.ag-grid.com/javascript-data-grid/component-cell-renderer/#reference-ICellRendererParams) and the defined `componentParams` from your `AppRenderer` definition.**
+
+### Utilizing a component in Elm
+
+To use a component for cell rendering, you can use the `AppRenderer { componentName : String, componentParams : Maybe Json.Encode.Value } (dataType -> String)` renderer in your column definition.
+
+```elm
+    { field = "details"
+    , renderer = AppRenderer { componentName = "linkRenderer", componentParams = Nothing } .url
+    , headerName = "Details"
+    , settings = { defaultSettings | editable = False }
```

Comment:
Should we enforce the `editable = False` by always setting it to this when the `AppRenderer` is used?

Reply:
No. You might still want to be able to edit the cell value normally. Even though you could still handle the editing in the component, I don't think we should enforce that.

Reply:
Right.